src/Components/Aspire.Microsoft.EntityFrameworkCore.Cosmos/EntityFrameworkCoreCosmosDBSettings.cs

```diff
@@ -50,4 +50,9 @@ public sealed class EntityFrameworkCoreCosmosDBSettings
     /// Gets or sets a string value that indicates what Azure region this client will run in.
     /// </summary>
     public string? Region { get; set; }
+
+    /// <summary>
+    /// Gets or sets the time in seconds to wait for the response to come back from the network peer.
+    /// </summary>
+    public int? RequestTimeout { get; set; }
```

Comment:
Since this one is already a "special case", what do you think about making this:

```suggestion
    public TimeSpan? RequestTimeout { get; set; }
```

That way we support the underlying data type.

Reply:
Makes sense.

Reply:
My concern with Timespan is the format in JSON, it's not obvious even for experienced devs what format you can use. How do you represent 10 seconds ? `10` (it's 10 days!!!), `00:10` (10 minutes) ...

And honestly I don't think someone would put something different than seconds, or maybe let's handle decimals in case someone wants milliseconds, and still accept the number of seconds to type `0.5` for 500ms

Reply:
I did have to figure out how to do this, so I agree with that concern.

Reply:
There are quite a few places that put TimeSpan in an options class, especially in dotnet/extensions:

* https://github.com/dotnet/extensions/blob/16b40667278d24a23af0c839f4c277499b3fee19/src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/LoggingOptions.cs#L72
* https://github.com/dotnet/extensions/blob/16b40667278d24a23af0c839f4c277499b3fee19/src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Latency/RequestLatencyTelemetryOptions.cs#L21
* https://github.com/App-vNext/Polly/blob/f99a4cea335ab7891ea6b56284b1d8e47f8e63ab/src/Polly.Core/Timeout/TimeoutStrategyOptions.cs#L24

We even have it in other places already in Aspire components:

https://github.com/dotnet/aspire/blob/46b39c260bd3346aa09a2814ad187bd646fc631f/src/Components/Aspire.Azure.Storage.Blobs/ConfigurationSchema.json#L83-L92

https://github.com/dotnet/aspire/blob/46b39c260bd3346aa09a2814ad187bd646fc631f/src/Components/Aspire.RabbitMQ.Client/ConfigurationSchema.json#L239-L243

https://github.com/dotnet/aspire/blob/46b39c260bd3346aa09a2814ad187bd646fc631f/src/Components/Aspire.StackExchange.Redis/ConfigurationSchema.json#L82-L86

We mark it as a "duration" in the JSON schema, so that should help customers figure out what format to use.

Reply:
I'm merging this PR. If we want to change this then I can open a follow up.